config-provisioning/src/main/java/com/yahoo/config/provision/ZoneEndpoint.java

```diff
@@ -27,16 +27,25 @@ public class ZoneEndpoint {
      * </ol>
      */
     public static final int generation = 0;
-    public static final ZoneEndpoint defaultEndpoint = new ZoneEndpoint(true, false, List.of());
-    public static final ZoneEndpoint privateEndpoint = new ZoneEndpoint(false, false, List.of());
+    public static final ZoneEndpoint defaultEndpoint = new ZoneEndpoint(true, false, false, List.of());
+    public static final ZoneEndpoint privateEndpoint = new ZoneEndpoint(false, false, false, List.of());
 
     private final boolean isPublicEndpoint;
     private final boolean isPrivateEndpoint;
+    private final boolean supportsTokenAuthentication;
```

Comment:
I think this should be `List<AuthMethod> authMethods` instead, to make it more extensible. The `defaultEndpoint` and `privateEndpoint` should have `List.of(AuthMethod.mtls)` as the default.

The getter/builders methods should then be named `authMethods/withAuthMethods`.